components/raftstore/src/store/fsm/apply.rs

```diff
@@ -3156,6 +3272,7 @@ pub struct Apply<C> {
     pub entries_size: usize,
     pub cbs: Vec<Proposal<C>>,
     pub bucket_meta: Option<Arc<BucketMeta>>,
+    pub wait_data: bool,
```

Comment:
Notify the change by a separate msg, see `Msg::LogsUpToDate` for reference

Reply:
Why? The `ApplyFsm` is never manipulated by multiple threads at the same time.

Reply:
I thought to share `waita_data` bewteen store thread (`PeerFsm`) and apply thread (`ApplyFsm`).

Because after the `wait_data` of `ApplyFsm` becomes `true`, the current PR implementation is always pending, so it cannot read the message will be sent by the `PeerFsm` thread after the apply snapshot is successful. 

The comment you mentioned below seems to have solved the problem, I'll think about it first.